core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -53,7 +55,7 @@
 
     public ScaledWriterScheduler(
             StageExecution stage,
-            Supplier<Collection<TaskStatus>> sourceTasksProvider,
+            Supplier<Collection<TaskInfo>> sourceTasksProvider,
```

Comment:
Is there any disadvantage to using TaskInfo instead of TaskStatus? 

IIRC `TaskInfo` refreshes at every 3 secs whereas `TaskStatus` at every 1 sec.

Reply:
TaskStatus does not contain `outputDataSize`. It is in `TaskStats` in `TaskInfo`.

Reply:
Yes, but if it's required we can put `outputDataSize` to `TaskStatus`. But I wonder if there's any problem using `TaskInfo` here?

Reply:
I don't think there is an issue using TaskInfo. Do you have anything in mind?

Reply:
I briefly looked into this. It seems fine to use TaskInfo.

Reply:
`TaskStatus` is fetched every 1s at most, while `TaskInfo` is fetched every 3s. 
I think it's fine to add output data size to `TaskStatus`. We should avoid bloating of `TaskStatus`, but one extra field is not a problem. We already have `physicalWrittenDataSize` there.